### PR TITLE
Fix issue with ABFSS & other file systems

### LINKED ISSUE #133: No FileSystem for scheme: abfss
Hi,

I`m trying to setup overwatch on ADLS Gen2 from my workspace and the setup fails in the Initializer with the following error:

```
IOException: No FileSystem for scheme: abfss
	at org.apache.hadoop.fs.FileSystem.getFileSystemClass(FileSystem.java:2660)
	at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:2667)
	at org.apache.hadoop.fs.FileSystem.access$200(FileSystem.java:94)
	at org.apache.hadoop.fs.FileSystem$Cache.getInternal(FileSystem.java:2703)
	at org.apache.hadoop.fs.FileSystem$Cache.get(FileSystem.java:2685)
	at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:373)
	at org.apache.hadoop.fs.Path.getFileSystem(Path.java:295)
	at com.databricks.labs.overwatch.utils.Helpers$.pathExists(Tools.scala:572)
	at com.databricks.labs.overwatch.pipeline.Initializer.dataTargetIsValid(Initializer.scala:265)
	at com.databricks.labs.overwatch.pipeline.Initializer.com$databricks$labs$overwatch$pipeline$Initializer$$validateAndRegisterArgs(Initializer.scala:198)
	at com.databricks.labs.overwatch.pipeline.Initializer$.apply(Initializer.scala:434)
```

spark.read.* & dbutils in Scala works from the same workspace/cluster with the same ADLS.

This forced me to create a mount point to workaround this.

### PR DESCRIPTION
**Pull request by alexott**
The problem was that:
* the `PipelineFunctions.cleansePathURI` function replaced the `//` with `/`, effectively breaking the urls like `abfss://container@storage_acct`
* We used empty Hadoop Configuration class instead of getting it from the SparkContext

To make it working correctly we need to set not only the  Spark configuration properties (like, `fs.azure.account`), but also the corresponding Hadoop properties (i.e. `spark.hadoop.fs.azure.account`)

This fixes #133

### CODE REVIEW

File: src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala [hunk 1]
```diff
@@ -603,8 +604,7 @@ object Helpers extends SparkSessionWrapper {
    */
   def parListFiles(path: String): Array[String] = {
     try {
-      val conf = new Configuration()
-      val fs = new Path(path).getFileSystem(conf)
+      val fs = new Path(path).getFileSystem(spark.sparkContext.hadoopConfiguration)
```
Comment (GeekSheikh) [inline src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala:607]: I have struggled with Serialization issues in the past. Is there a specific reason you moved this other than solely code cleanliness? Note the usage inside the lamba [HERE](https://github.com/databrickslabs/overwatch/blob/993ea86e956e7e01453fc58de33e2543392ea957/src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala#L672).

Did you test this with sparkEvents module to ensure spark events could be loaded without throwing `task not serializable` errors?

File: src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala [hunk 2]
```diff
@@ -633,8 +633,16 @@ object Helpers extends SparkSessionWrapper {
...
   def globPath(path: String, fromEpochMillis: Option[Long] = None, untilEpochMillis: Option[Long] = None): Array[PathStringFileStatus] = {
+    globPath(path, spark.sparkContext.hadoopConfiguration, fromEpochMillis, untilEpochMillis)
+  }
+
+  def globPath(path: String, conf: SerializableConfiguration, fromEpochMillis: Option[Long],
+               untilEpochMillis: Option[Long]): Array[PathStringFileStatus] = {
+    globPath(path, conf.value, fromEpochMillis, untilEpochMillis)
+  }
+
+  def globPath(path: String, conf: Configuration, fromEpochMillis: Option[Long], untilEpochMillis: Option[Long]): Array[PathStringFileStatus] = {
```
Comment (GeekSheikh) [inline src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala:644]: Same [here](https://github.com/databrickslabs/overwatch/blob/993ea86e956e7e01453fc58de33e2543392ea957/src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala#L677) as it is also used inside the lamda. 

I like your code better, and I tried several ways but kept getting serializable issues. Could have been REPL related issues but just wanted to make sure you tested it.

File: src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala [hunk 3]
```diff
@@ -865,20 +878,20 @@ object Helpers extends SparkSessionWrapper {
    *                 globs will be dropped in parallel
    */
   private[overwatch] def fastrm(topPaths: Array[String]): Unit = {
+    val conf = new SerializableConfiguration(spark.sparkContext.hadoopConfiguration)
```
Comment (GeekSheikh) [inline src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala:881]: same here as above

**Review by GeekSheikh — COMMENTED** [review]
Did you do end-to-end testing for all modules on AWS and Azure?

Please rev the version to 4.13. There also needs to be a corresponding PR to public_docs branch to add change notes to the public facing [CHANGELOG](https://databrickslabs.github.io/overwatch/changelog/)

**Comment by GeekSheikh** [comment]
For future reference, please link the issue to the PR that resolves it. I linked it this time.

**Comment by GeekSheikh** [comment]
> Did you do end-to-end testing for all modules on AWS and Azure?
> 
> Please rev the version to 4.13. There also needs to be a corresponding PR to public_docs branch to add change notes to the public facing [CHANGELOG](https://databrickslabs.github.io/overwatch/changelog/)

nevermind on revving the build version. I forgot you were going to dev...that'll be 4.2 before it goes to master so no matter.

File: src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala (hunk 1 above)
Comment (GeekSheikh) [reply src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala:607]: if you don't have time to test, I will be testing it as part of the larger 0.4.2 release testing.

File: src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala (hunk 3 above)
Comment (alexott) [reply src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala:421]: fixed

File: src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala (hunk 1 above)
Comment (alexott) [reply src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala]: yes, I tested with all modules, include `sparkEvents`

File: src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala (hunk 2 above)
Comment (alexott) [reply src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala:185]: fixed

**Review by GeekSheikh — APPROVED** [review]
I ran full gammit of pipeline tests and had no issues. Please note to anyone using this develop branch that this hasn't been fully tested for data quality yet. I am merging to develop but develop is in development. I suggest you use this pipeline to enable and test setups but I suggest that all data be dropped and rebuilt after the 4.2 official release or there may be unexpected challenges.

**Pull request merged**